Add a way to populate the kubeconfig using a keytone token for qbertconfig

k2mahajan opened the pull request:
Added optional cli argument '-t' or '--token' for keystone token. The toke, if passed, is stored in kubeconfig file with the key 'token'. If not passed, the previous behavior holds where username and password are stored as base64 encoded string as 'token'.

Testing:
Tested manually as follows (with a modified main.py as an executable):
[root@ketan-latest.platform9.horse qbertconfig(service_admin)]# python qbertconfig/cli/main.py fetch --uuid fc0c4a3b-bc3e-4668-a1af-8b801dee171b -k k.cfg **-t gAAAAABczCvxFQ1k9UGy6vkcUVBiQazRqb-6CYQ51J282NwkZYdrplmEZe7bm5Vc3T4dvlUgrU9qwNX_6dAzN4ry_bxbeqYWVJY_wZTQ5MSdOI4uqWMnctUAPygHRkG84Zkk03hUOYpdmIWNost0R6247K9WM0MJ7GROjMnQnq68qxWubXtPVQ0**
2019-05-03 11:55:06,961 - qbertconfig.QbertClient - INFO - Getting kubeconfig for cluster 'cniflannel' (fc0c4a3b-bc3e-4668-a1af-8b801dee171b)
[root@ketan-latest.platform9.horse qbertconfig(service_admin)]# cat k.cfg
.
.
.
- name: localhost:8080-ketan@platform9.com
  user: {token: **gAAAAABczCvxFQ1k9UGy6vkcUVBiQazRqb-6CYQ51J282NwkZYdrplmEZe7bm5Vc3T4dvlUgrU9qwNX_6dAzN4ry_bxbeqYWVJY_wZTQ5MSdOI4uqWMnctUAPygHRkG84Zkk03hUOYpdmIWNost0R6247K9WM0MJ7GROjMnQnq68qxWubXtPVQ0**}

Issues:
https://platform9.atlassian.net/browse/PMK-1738

agrahamlincoln commented:
The user has already provided Openstack auth credentials, so providing a token is unnecessary and adds an extra step for the end-user. My thought is the `-t` flag could be a switch to decide on whether we use the credentials or a token. IMO, the default behavior should be to use a token instead of credentials, given that it is more secure.

I think it would be better to re-use the token from QbertClient's session adapter (`keystoneauth1.session.Session`) similarly to how the credentials are accessed.

We could also request a new token from keystone, but this isn't as straightforward due to the [features of keystoneauth1](https://docs.openstack.org/keystoneauth/latest/using-sessions.html#features):

> Clients are not expected to have any knowledge of an identity token or any other form of identification credential. Service, endpoint, major version discovery, and microversion support discovery are handled by the Session and plugins

agrahamlincoln commented:
Looks good!

A couple more things:
- Can you set your Git username so your commits aren't coming from `root`? Some instructions here: https://help.github.com/en/articles/setting-your-username-in-git
- Please fix any merge conflicts so this can be fast-forward merged. Sorry about that, I pushed some python3 compatibility stuff earlier that week that looks to conflict a little.

Finally a note about the default behavior:
It looks like the default functionality is still going to use the base64-encoded credentials. I'm not sure if anyone else has opinions on this, but I think we should default to using the token as the default behavior. I'll leave naming and such to you, but I think we should name the top level parameter to `--encode-creds` or as a flag `-c` and the default value should be False so the token is used if this flag is not specified.

k2mahajan commented:
Thanks for the suggestions, @agrahamlincoln . I have changed the default behavior. Please review. I had fixed the merged conflicts yesterday, so should be good now.
